stdlib/src/builtin/string_literal.mojo

```diff
@@ -333,7 +333,7 @@ struct StringLiteral(
         return ptr.bitcast[UInt8]()
 
     @always_inline
-    # FIXME(MSTDL-956): This should return a pointer with StaticConstantLifetime.
```

Comment:
**Question**: how are we going to support this in the future?

I don't think there's a *clean* trait based way to sometimes return `UnsafePointer[lifetime=StaticConstantLifetime]` and other times `UnsafePointer[lifetime=__lifetime_of(self)]. We could use of associated types but that feels a bit excessive.

The bitcast overload is fine because we can just make that a generic function.

Reply:
We could always worry about this later though.

Reply:
Please keep the FIXME.  We need an ability to specify immutable ("const") pointers, which is what this is hinting at.  We have that machinery in place now to go explore that (pointer carries an origin parameter now).  This enables us to plumb pointers with an associated origin/mutability through the `unsafe_ptr()` methods so it can be tracked in the origin checking in the compiler.  If you have any interest in exploring this work, feel free to ping me and I can share more details.